### DISCUSSION TITLE
Netplan For Customized Ubuntu 23 Server

### DISCUSSION THREAD

**Discussion by ozgurgunesulkesi** [cockpit-project/cockpit#19282] [Q&A] [answered]
### Explain what happens

I have a question. I try to make a customized ubuntu server img file for raspberry with cockpit preinstalled and a netplan like this one:
network:
  version: 2
  renderer: NetworkManager
  ethernets:
    eth0:
      dhcp4: no
      addresses: [10.10.10.100/24]
      gateway4: 10.10.10.1
      nameservers:
        addresses: [8.8.8.8, 8.8.4.4] 
for testing now.  But what i encouter is i try to change the IP afterward using cockpit. But when i reboot the IP goes back to 10.10.10.100.  Can someone tell why this happens or how i can setup a static ip for eth0 preconfigured, which can later be changed?

### Where is the problem in Cockpit?

None

### Server operating system

None

**Comment by martinpitt** [comment, marked as answer]
It happens because netplan generates its configuration in /run/NetworkManager/, which overrides /etc/NetworkManager (where NM stores its configuration that is set via Cockpit or `nmcli`).

In other words: Cockpit cannot edit the netplan config, only direct NM config. So this isn't supported.

**Reply by matthewGEA** [reply]
@martinpitt  Does this statement published after this Github discussion reveal any options going forward?

"Activating the bidirectional NetworkManager-Netplan integration allows for any configuration change made through NetworkManager to be propagated back into Netplan. Changes made in Netplan itself will still be visible in NetworkManager, as before. This way, Netplan can be considered the “single source of truth” for network configuration across all variants of Ubuntu, with the network configuration stored in /etc/netplan/, using Netplan’s common and declarative YAML format."
https://ubuntu.com/blog/netplan-configuration-across-desktop-server-cloud-and-iot

**Reply by martinpitt** [reply]
@matthewGEA nice! So this should start to work by and large automatically. Cockpit's Networking page does not write any files, it only talks to NM via its D-Bus API. If that gets reflected into netplan config files, so much the better!
